### PR TITLE
Make realtime server upload timeout configurable

### PR DESCRIPTION
**Pull request by npawar**

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ public String segmentConsumed(@QueryParam(SegmentCompletionProtocol.PARAM_INSTAN
 
     SegmentCompletionProtocol.Response response = _segmentCompletionManager.segmentConsumed(requestParams);
     final String responseStr = response.toJsonString();
-    LOGGER.info("Response to segmentConsumed:{}", responseStr);
+    LOGGER.info("Response to segmentConsumed for segment {} is :{}", segmentName, responseStr);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java]: Suggest changing the message to: "Response to segmentConsumed for segment: '{}' is: {}"
(Add single quote around segment name so we know if there are leading or trailing spaces, also for the ':', no space before and one space after)

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 2]
```diff
@@ -158,7 +158,7 @@ public String segmentStoppedConsuming(@QueryParam(SegmentCompletionProtocol.PARA
     SegmentCompletionProtocol.Response response =
         _segmentCompletionManager.segmentStoppedConsuming(requestParams);
     final String responseStr = response.toJsonString();
-    LOGGER.info("Response to segmentStoppedConsuming:{}", responseStr);
+    LOGGER.info("Response to segmentStoppedConsuming for segment {} is:{}", segmentName, responseStr);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java]: "Response to segmentStoppedConsuming for segment: '{}' is: {}"

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 3]
```diff
@@ -188,7 +188,7 @@ public String segmentCommitStart(@QueryParam(SegmentCompletionProtocol.PARAM_INS
     SegmentCompletionProtocol.Response response =
         _segmentCompletionManager.segmentCommitStart(requestParams);
     final String responseStr = response.toJsonString();
-    LOGGER.info("Response to segmentCommitStart:{}", responseStr);
+    LOGGER.info("Response to segmentCommitStart for segment {} is :{}", segmentName, responseStr);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java]: "Response to segmentCommitStart for segment: '{}' is: {}"

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 4]
```diff
@@ -236,7 +236,7 @@ public String segmentCommitEnd(@QueryParam(SegmentCompletionProtocol.PARAM_INSTA
     SegmentCompletionProtocol.Response response = _segmentCompletionManager
         .segmentCommitEnd(requestParams, isSuccess, isSplitCommit, committingSegmentDescriptor);
     final String responseStr = response.toJsonString();
-    LOGGER.info("Response to segmentCommitEnd:{}", responseStr);
+    LOGGER.info("Response to segmentCommitEnd for segment {} is:{}", segmentName, responseStr);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java]: "Response to segmentCommitEnd for segment: '{}' is: {}"

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 5]
```diff
@@ -362,7 +362,7 @@ public String segmentUpload(@QueryParam(SegmentCompletionProtocol.PARAM_INSTANCE
 
       String response = new SegmentCompletionProtocol.Response(responseParams).toJsonString();
 
-      LOGGER.info("Response to segmentUpload:{}", response);
+      LOGGER.info("Response to segmentUpload for segment {} is:{}", segmentName, response);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java]: "Response to segmentUpload for segment: '{}' is: {}"

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 6]
```diff
@@ -413,7 +413,7 @@ public String segmentCommitEndWithMetadata(@QueryParam(SegmentCompletionProtocol
         .segmentCommitEnd(requestParams, isSuccess, isSplitCommit,
             CommittingSegmentDescriptor.fromSegmentCompletionReqParamsAndMetadata(requestParams, segmentMetadata));
     final String responseStr = response.toJsonString();
-    LOGGER.info("Response to segmentCommitEnd:{}", responseStr);
+    LOGGER.info("Response to segmentCommitEnd for segment {} is:{}", segmentName, responseStr);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java]: "Response to segmentCommitEndWithMetadata for segment: '{}' is: {}"
(Make it different from segmentCommitEnd)

File: pinot-core/src/main/java/org/apache/pinot/server/realtime/ServerSegmentCompletionProtocolHandler.java [hunk 7]
```diff
@@ -40,7 +40,8 @@
  */
 public class ServerSegmentCompletionProtocolHandler {
   private static Logger LOGGER = LoggerFactory.getLogger(ServerSegmentCompletionProtocolHandler.class);
-  private static final int SEGMENT_UPLOAD_REQUEST_TIMEOUT_MS = 30_000;
+  private static final int DEFAULT_SEGMENT_UPLOAD_REQUEST_TIMEOUT_MS = 300_000;
+  private static final String CONFIG_OF_SEGMENT_UPLOAD_REQUEST_TIMEOUT_MS = "upload.request.timeout.ms";
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/server/realtime/ServerSegmentCompletionProtocolHandler.java]: Can you move these into CommonConstants?

**Comment by npawar** [comment]
@Jackie-Jiang fixed the `:` and `space` and moved config keys to CommonConstants. I prefer not to add single quotes around segment. I believe the segment name cannot have trailing or leading spaces in that particular place. 
@jenniferdai the offline segment upload timeouts will be configured at the client applications which are responsible for pushing segments. I didn't see any such config or defaults on our end. Also, I prefer if we keep the realtime upload times separate, as we are more in control of the segment end-to-end, as compared to offline. Does that work?

**Pull request merged**